cmd/restic/cmd_dump.go

```diff
@@ -194,28 +212,28 @@ func getNodeData(ctx context.Context, output io.Writer, repo restic.Repository,
 	return nil
 }
 
-func tarTree(ctx context.Context, repo restic.Repository, rootNode *restic.Node, rootPath string) error {
+func tarTree(ctx context.Context, repo restic.Repository, rootNode *restic.Node, rootPath string, tw *tar.Writer) error {
 
 	if stdoutIsTerminal() {
 		return fmt.Errorf("stdout is the terminal, please redirect output")
 	}
 
-	tw := tar.NewWriter(os.Stdout)
-	defer tw.Close()
-
 	// If we want to dump "/" we'll need to add the name of the first node, too
 	// as it would get lost otherwise.
 	if rootNode.Path == "/" {
 		rootNode.Path = path.Join(rootNode.Path, rootNode.Name)
 		rootPath = rootNode.Path
 	}
 
-	// we know that rootNode is a folder and walker.Walk will already process
-	// the next node, so we have to tar this one first, too
 	if err := tarNode(ctx, tw, rootNode, repo); err != nil {
 		return err
 	}
 
+	// If this is a file we are finished
+	if rootNode.Subtree == nil {
```

Comment:
That's not the correct way to test for a tree, use this:
```suggestion
	if rootNode.Type != "tree" {
```

Reply:
Hmmm... the goal is rather to test for a file.

So I guess:
```
if rootNode.Type == "file"
```

Would be the best solution here? As your suggestion would also return if the node is a folder or a link.